docs/debug.md

```diff
@@ -0,0 +1,173 @@
+# Debug Guide
```

Comment:
Make this explicit with something like "Debugging the Azure Provider with VS Code and the Remote Container Exntension"